app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import androidx.core.app.NotificationChannelCompat
+import androidx.core.app.NotificationManagerCompat
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+
+@RunWith(FenixRobolectricTestRunner::class)
+class NotificationManagerCompatTest {
+
+    private lateinit var tested: NotificationManagerCompat
+
+    @Before
+    fun setup() {
+        tested = mockk(relaxed = true)
+
+        mockkStatic("org.mozilla.fenix.ext.NotificationManagerCompatKt")
+    }
+
+    @Test
+    fun `areNotificationsEnabledSafe returns false when areNotificationsEnabled throws an exception`() {
+        every { tested.areNotificationsEnabled() } throws RuntimeException()
+
+        assertEquals(false, tested.areNotificationsEnabledSafe())
+    }
+
+    @Test
+    fun `areNotificationsEnabledSafe returns false when areNotificationsEnabled returns false`() {
+        every { tested.areNotificationsEnabled() } returns false
+
+        assertEquals(false, tested.areNotificationsEnabledSafe())
+    }
+
+    @Test
+    fun `areNotificationsEnabledSafe returns true when areNotificationsEnabled returns true`() {
+        every { tested.areNotificationsEnabled() } returns true
+
+        assertEquals(true, tested.areNotificationsEnabledSafe())
+    }
+
+    @Test
+    fun `isNotificationChannelEnabled returns true when getNotificationChannelCompat returns a channel and areNotificationsEnabled returns true`() {
```

Comment:
should we also have a test for when the channel importance is NotificationManagerCompat.IMPORTANCE_NONE and a test for when the build version is < or = to Build.VERSION_CODES.O ?

Reply:
Good point, updated to have `IMPORTANCE_NONE` test case 👍 
I was checking how to fake a different OS version, roboelectric has some ways to config the sdk used, which would run the tests in a different environment/config. Having an abstraction like BuildVersionProvider and fake in test might work but seemed a bit much for just this use case 😅  do you know have any ideas how to go about this?

Reply:
Added the test using `ReflectionHelpers` and a follow up ticket for `BuildVersionProvider` we discussed to improve this --> https://bugzilla.mozilla.org/show_bug.cgi?id=1810179